tests/cases/py_client/lib_test.py

```diff
@@ -438,6 +438,17 @@ def processEvent(event):
         file = self.client.uploadFileToItem(item['_id'], testPath)
         self.assertEqual(file['mimeType'], 'text/plain')
 
+        # Test uploading to a folder
+        self.client.uploadFileToFolder(
+            str(self.publicFolder['_id']), path, reference='test4_reference')
+        while (not events.daemon.eventQueue.empty() and
+                time.time() - starttime < 5):
+            time.sleep(0.05)
```

Comment:
Looking at this loop condition which you copied from above, it's not really what we want since it suffers from the ABA problem. We should really be waiting on the state of `eventList`. Could you change it in all the places it's used, or simply use the synchronous event everywhere?